source/deployment/permissions-backend.rst

```diff
@@ -0,0 +1,104 @@
+
+Schemes
+~~~~~~~~~
+
+Schemes describe the default roles applied to users in a context, and all child contexts. Schemes are either defined specifically for a context, or if they are not specified, the relevant parts of the parent context’s scheme are applied, ultimately climbing the hierarchy to the System Scheme, which serves the purpose of providing the system-wide defaults. For example, if Team A does not have a team-scoped scheme defined, the System Scheme will provide the defaults for all contexts in Team A.
+
+Additionally, the lowest-scoped scheme always takes precedence in the context. For example, if Team B has a team-scoped scheme, that scheme takes precedence over the System Scheme defaults for all contexts in Team B. 
+
+Data Structure
+----------------
+
+Permissions
+~~~~~~~~~~~~
+
+Permissions in Mattermost are a property of the server code base and are not created or modified dynamically. The current set of permissions are as described in the table below.
+
+XXXXXX GG: Please review/tweak this and then I'll create the RST table to put here: https://docs.google.com/spreadsheets/d/1z3yUIQs2JaxmUjD5Q8MZSijvbE2o8CoHdvV5Vfj-y4Y/edit#gid=0 
+
+Applicable database fields and tables.
+
+``User`` ``TeamMember`` ``ChannelMember`` Tables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Roles are applied to objects that represents that user’s membership in a context. These are represented by the following objects in the Mattermost data model:
+
+- ChannelMember for a channel context.
+- TeamMember for a team context.
+- User for the system context. 
+
+XXXXXX GG: can we expand on this, what fields are relevant in these tables?
+
+``Roles`` Table
+~~~~~~~~~~~~~~~~
+
+Roles are dynamic and user configurable, necessitating a database table with the following fields:
+
+- Id (Autoincrement, Primary Key)
+- Name (Unique String with Character Constraints, e.g. “team_user”).
+- Display Name (String)
+- Description (String)
+- Permissions (String - golang style “fields” with permission IDs).
+- Scheme Managed (bool) - indicates whether this role is managed as part of a scheme.
+
+``Schemes`` Table
+~~~~~~~~~~~~~~~~~~
+
+Schemes are dynamic and user configurable, necessitating a database table with the following fields:
+
+- Id (Autoincrement, Primary Key)
+- Name (Unique String with Character Constraints, e.g. “corporate_scheme”)
+- Description (String)
+- Scope (String): Team or Channel
+- Team Admin Role (String): Empty if Channel Scope
+- Team User Role (String): Empty if Channel Scope
+- Channel Admin Role (String): Always provided
+- Channel User Role (String): Always provided
+
+XXXXXX GG: What about the system admin and system user roles for the system default scheme? How are those defined if the schemes table doesn't have those fields?
```

Comment:
The system scheme doesn't exist in the table - only custom schemes. System scheme is built into the product and it's roles are defined as `BuiltIn: true` in the `Roles` table.